pixeldata/src/attribute.rs

```diff
@@ -152,20 +153,101 @@ pub fn pixel_data<D: DataDictionary + Clone>(
         .context(MissingRequiredSnafu { name })
 }
 
+fn get_from_shared<D: DataDictionary + Clone>(obj: &FileDicomObject<InMemDicomObject<D>>, selector: [Tag; 2]) -> Option<Vec<&InMemElement<D>>> {
+    obj.element(tags::SHARED_FUNCTIONAL_GROUPS_SEQUENCE).ok()
```

Comment:
`obj.get` is a more appealing method here. It was introduced some time after `.element` and takes advantage of the fact that element retrieval never fails when it exists.

```suggestion
    obj.element(tags::SHARED_FUNCTIONAL_GROUPS_SEQUENCE).ok()
```

Since this function returns an `Option<_>`, you can chain `Some` contents using the `?` operator. For example:

```rust
let e = obj.get(tags::SHARED_FUNCTIONAL_GROUPS_SEQUENCE)?;
let item = e.items()?.get(0)?;
// ...
```

Although it's worth noting that some parts of the code may be hiding input data errors (`to_float64().ok()` means that we would have no way of knowing whether the value could not be converted to `f64` or the value was missing). `Options` and `Result` of any kind can be harmonized together into a `Result` with a specific error type through Snafu.

Reply:
Leaving this open as I implemented the suggestions, but still left the `get_from_shared` and `get_from_per_frame` returning `Option`s instead of `Result`s.  For  `voi_lut_function` there are a few specific called out errors, but for the  others `rescale_slope` /`rescale_intercept`/`window_center`/`window_width` I've left them just basically just returning Options with a default value if there is any error looking up or converting.  Not sure if you want specific errors called out there